k8s/upgrade/src/preflight_validations.rs

```diff
@@ -0,0 +1,20 @@
+use crate::user_prompt;
+use operators::upgrade::{common::error, controller::reconciler};
+
+/// Validation to be done before applying upgrade.
+pub async fn preflight_check() -> Result<(), error::Error> {
+    console_logger::info(user_prompt::UPGRADE_WARNING);
+    match reconciler::is_rebuilding().await {
+        Ok(is_rebuilding) => {
+            if is_rebuilding {
+                console_logger::warn(user_prompt::REBUILD_WARNING, "");
+                std::process::exit(1);
```

Comment:
Shouldn't we return an error instead of exit?

Reply:
Its not an error . It just writes to console and exit.

Reply:
What I mean is shouldn't we return an error and then the code handling the error decide to exit?
Seems like we have a bunch of process exit all over the place ...